contracts/contracts/GameItems.sol

```diff
@@ -1,32 +1,223 @@
-     * @dev See {IERC165-supportsInterface}.
-     */
-    function supportsInterface(bytes4 interfaceId)
-        public
-        view
-        virtual
-        override(AccessControl, ERC1155)
-        returns (bool)
-    {
+        return _items;
+    }
+
+    /// @notice Retreives the current number of players/holders
+    function getNumberPlayers() override external view returns (uint256) {
+        return numberPlayers;
+    }
+
+    /// @notice Retreives Unlock Date
+    /// @dev Can be deprecated
+    /// @return uint256 time in epoch * 1000
+    function getUnlockDate() external view returns (uint256) {
+        return unlockDate;
+    }
+
+    function initialMint(address receiver) override external onlyMinter {
+        //// TESTING VALUE === 6
+        require(numberItems == 6, CONTRACT_IN_ACTIVE); /// PRODUCTION VALUE === require(numberItems == 12, CONTRACT_IN_ACTIVE);
+        require(_noBalances(receiver), INVALID_NEW_PLAYER);
+        uint256 _rng = _getRandomNumber();
+        uint256[3] memory tokenIds = _rng == 0 ? [uint256(0), 2, 4] : [uint256(1), 3, 5];
+        for (uint256 i = 0; i < 3; i++) {
+            _internalMint(receiver, tokenIds[i]);
+        }
+        numberPlayers++;
+        emit NewPlayer(receiver, tokenIds, block.timestamp);
+    }
+
+    /// @notice Checks if contract is locked
+    /// @dev Internal/External
+    /// @dev Returns True if Locked
+    /// @return bool if is locked
+    function isLocked() public view returns (bool) {
+        return block.timestamp <= unlockDate;
+    }
+
+    /// @notice Required for Inheriting Contracts
+    function supportsInterface(bytes4 interfaceId) public view override (AccessControlEnumerable, ERC1155, IERC165) returns (bool) {
         return super.supportsInterface(interfaceId);
     }
 
+    /// @notice Internal Mint
+    /// @dev Mints the Selected Token Id to the Selected Reciever
+    /// @dev Always mints only 1
+    /// @param receiver address of the person who the token will go to
+    /// @param tokenId The Specific token they should receive
+    function _internalMint(address receiver, uint256 tokenId) internal {
+        _mint(receiver, tokenId, 1, "");
+    }
+
+    /// @notice Checks If a Player being added has toknes
+    /// @dev Will Return True -> Allowing New Player to Be Created
+    /// @dev If Player has any token, will not allow them to be added again
+    /// @param checkFor address to check
+    function _noBalances(address checkFor) internal view returns (bool) {
+        for (uint256 i = 0; i < numberItems; i++) {
+            if (balanceOf(checkFor, i) > 0) return false;
+        }
+        return true;
+    }
+
+    /// @notice SKALE RNG Casting
+    function _getRandomNumber() internal view returns (uint256) {
+        uint256 _rng = uint256(_randomNumberGenerator()) % 1000;
+        return _rng < 500 ? 1 : 0;
+    }
+
+    /// @notice SKALE RNG (CHECK!)
+    function _randomNumberGenerator() internal view returns (bytes32 addr) {
```

Comment:
I think to facilitate testing this should probably be in another contract, but not gonna block on this

Reply:
I will not argue there, for the first iteration, the core unit testing I have always found to be sufficient for an initial build as I can build the tests and then confirm the core contracts there. 

Can look at doing some mocks for this.